packages/cdktf/lib/app.ts

```diff
@@ -63,37 +73,66 @@ export class App extends Construct {
 
     this.loadContext(options.context);
 
-    const node = Node.of(this);
+    const node = this.node;
     if (options.stackTraces === false) {
-      node.setContext(ConstructMetadata.DISABLE_STACK_TRACE_IN_METADATA, true);
+      node.setContext(DISABLE_STACK_TRACE_IN_METADATA, true);
     }
 
     node.setContext("cdktfVersion", version);
+
+    if (!fs.existsSync(this.outdir)) {
+      fs.mkdirSync(this.outdir);
+    }
+    this.manifest = new Manifest(version, this.outdir);
   }
 
   /**
    * Synthesizes all resources to the output directory
    */
   public synth(): void {
-    if (!fs.existsSync(this.outdir)) {
-      fs.mkdirSync(this.outdir);
-    }
-
-    const manifest = new Manifest(version, this.outdir);
-
-    Node.of(this).synthesize({
+    const session: ISynthesisSession = {
       outdir: this.outdir,
       skipValidation: this.skipValidation,
-      sessionContext: {
-        manifest,
-      },
+      manifest: this.manifest,
+    };
+
+    invokeAspects(this);
+
+    const stacks = this.node
+      .findAll()
+      .filter<TerraformStack>(
+        (c): c is TerraformStack => c instanceof TerraformStack
+      );
+
+    // collect Annotations into Manifest
+    stacks.forEach((stack) => {
+      const annotations = stack.node
+        .findAll()
+        .map((node) => ({
+          node,
+          metadatas: node.node.metadata.filter(isAnnotationMetadata),
+        }))
+        .map<StackAnnotation[]>(({ node, metadatas }) =>
+          metadatas.map((metadata) => ({
+            constructPath: node.node.path,
+            level: metadata.type as AnnotationMetadataEntryType,
+            message: metadata.data,
+            stacktrace: metadata.trace,
+          }))
+        )
+        .reduce((list, metadatas) => [...list, ...metadatas], []); // Array.flat()
+
+      // it is readonly but this is the place where we are allowed to write to it
+      (this.manifest.forStack(stack).annotations as any) = annotations;
```

Comment:
Could you elaborate on why the and annotations are collected on app level, rather than having the stacks append the data to the manifest?

Reply:
There's no particular reason to it. But I think you're right: It might be better suited to move this to the `StackSynthesizer` instead.
At the point I wrote this, `StackManifest`s had this [weird behavior](https://github.com/hashicorp/terraform-cdk/blob/d904505481c7daee448b531572b2194f79b4e06f/packages/cdktf/lib/manifest.ts#L37) of being collected in an Array that could also end up with multiple ones of the same (which were reduced when rendering), so I had to make sure I put the annotations on the last `manifest.forStack(stack)` so it would end up as the last item to be reduced and [end up in the result](https://github.com/hashicorp/terraform-cdk/blob/d904505481c7daee448b531572b2194f79b4e06f/packages/cdktf/lib/manifest.ts#L47).